src/stream/src/executor/source.rs

```diff
@@ -152,37 +152,56 @@ impl SourceReader {
         mut stream_reader: Box<dyn StreamSourceReader>,
         notifier: Arc<Notify>,
         expected_barrier_latency_ms: u64,
-        mut inject_source_rx: UnboundedReceiver<Box<SourceStreamReaderImpl>>,
+        mut inject_source_rx: Receiver<Box<SourceStreamReaderImpl>>,
     ) {
-        'outer: loop {
-            let now = Instant::now();
-
-            // We allow data to flow for `expected_barrier_latency_ms` milliseconds.
-            while now.elapsed().as_millis() < expected_barrier_latency_ms as u128 {
-                if let Ok(reader) = inject_source_rx.try_recv() {
-                    stream_reader = reader;
-                }
-                match stream_reader.next().await {
-                    Ok(chunk) => yield chunk,
-                    Err(e) => {
-                        // TODO: report this error to meta service to mark the actors failed.
-                        error!("hang up stream reader due to polling error: {}", e);
-
-                        // Drop the reader, then the error might be caught by the writer side.
-                        drop(stream_reader);
-                        // Then hang up this stream by breaking the loop.
-                        break 'outer;
+        let (msg_tx, mut msg_rx) = tokio::sync::mpsc::channel::<Result<StreamChunkWithState>>(std::mem::size_of::<Result<StreamChunkWithState>>());
```

Comment:
We should specify the item count here instead of memory size.

Reply:
fixed